api/src/services/xml_generation/header_generator.py

```diff
@@ -97,24 +134,75 @@ def _format_date(self, date_value: date | None) -> str | None:
             return None
         return date_value.isoformat()
 
-    # TODO: this is likely going to be changed based on additional research into how the hash value is calculated
+    def _get_received_datetime(self) -> str | None:
+        """Get the ReceivedDateTime from submitted_at or current time in ISO 8601 format."""
+        timestamp = self.application.submitted_at if self.application.submitted_at else utcnow()
+        return timestamp.isoformat(timespec="milliseconds")
+
+    def _get_submitter_name(self) -> str:
+        """Get the submitter name from the submitted_by user.
+
+        Returns first_name + last_name if available, otherwise email, or 'unknown' if no user.
+        """
+        if not self.application.submitted_by:
+            return "unknown"
+
+        user = self.application.submitted_by_user
+        if not user:
+            return "unknown"
+
+        if user.profile and user.profile.first_name and user.profile.last_name:
+            return f"{user.profile.first_name} {user.profile.last_name}"
```

Comment:
First/last name aren't nullable, if they have a profile, they'll have a first/last name.

```suggestion
        if user.profile:
            return f"{user.profile.first_name} {user.profile.last_name}"
```